Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -525,13 +525,33 @@ CGImageRef CGImageCreateCopyWithColorSpace(CGImageRef ref, CGColorSpaceRef color
 }
 
 /**
- @Status Stub
+ @Status Interoperable
+ @Notes This function does not defer the composition of its mask.
+        This function supports non-grayscale alpha masks, unlike the reference platform.
  */
 CGImageRef CGImageCreateWithMask(CGImageRef image, CGImageRef mask) {
-    // TODO  #1425 - masks are applied during rendering via D2D.
     RETURN_NULL_IF(!image);
-    UNIMPLEMENTED();
-    return StubReturn();
+    RETURN_NULL_IF(!mask);
+    RETURN_NULL_IF(CGImageIsMask(image));
```

Comment:
Ops, LOL I was thinking it was mask.

Can you add a check to see if the mask is really as mask?

---
In reply to: [92708221](https://github.com/Microsoft/WinObjC/pull/1555#discussion_r92708221) [](ancestors = 92708221)